doc/bucket-class-crd.md

```diff
@@ -1,73 +1,100 @@
+  - Each rule is an object containing a `rule_id`, a `destination bucket`, and an optional `filter` key that contains a `prefix` field.
+  - When a filter with prefix is provided - only objects keys that match the prefix will be replicated
 
-Namespace policy of type cache will require the following configuration:
-  - Hub Resource - a single namespace-store, defines the read and write target of the namespace bucket.
-  - TTL - defines the TTL of the cached data.
+Replication policy:
 
+A bucket-class will define a replication policy for all future namespace buckets who utilize it.
+The policy is a JSON-compliant array of rules.
+  - Each rule is an object that contains the following keys:
+    - `rule_id` - which identifies the rule
+    - `destination_bucket` - which dictates the target NooBaa buckets that the objects will be copied to
+    - (optional) `{"filter": {"prefix": <>}}` - if the user wishes to filter the objects that are replicated, the value of this field can be set to a prefix string
+    - (optional, log-based optimization, see below) `sync_deletions` - can be set to a boolean value to indicate whether deletions should be replicated
+    - (optional, log-based optimization, see below) `sync_versions` - can be set to a boolean value to indicate whether object versions should be replicated
 
-Replication policy:
-A bucket-class will define a replication policy when an admin would like to replicate objects 
-within noobaa bucket to another noobaa bucket (source/destination buckets can be regular/namespace buckets).
-
-Replication policy will require the content of a JSON file which defines array of rules.
-  - Each rule is an object contains rule_id, destination bucket and optional filter object that contains prefix field.
-  - When a filter with prefix is defined - only objects keys that match prefix, will be replicated.
-
-Constraints:
-A backing-store name may appear in more than one bucket-class but may not appear more than once in a single bucket-class.
-The operator cli currently only supports a single tier placement-policy for a bucket-class.
-Upon creating regular buckets, the user will first need to create a placement-bucketclass which contains placemant policy.
-Upon creating namespace buckets, the user will first need to create a namespace-bucketclass which contains namespace policy.
-A namespace bucket class of type cache must contain both Placement-policy or Namespace-policy.
-A namespace bucket class of type single/multi must contain Namespace-policy.
-YAML must be used to create a bucket-class with a placement-policy that has multiple tiers.
-Placement-policy is case sensitive and should be of value (Mirror|Spread).
-Namespace-policy is case sensitive.
-
-For more information on using bucket-classes from S3 see [S3 Account](s3-account.md).
+In addition, each policy can be set to optimize replication by utilizing logs (configured and supplied by the user):
```

Comment:
I think we need to clarify that this optimization ("log-based replication") is only on AWS S3 and Azure Blob storage.